recipes/ovs_offload/1_virt_ovs_vxlan.py

```diff
@@ -54,11 +54,13 @@ def do_pings():
 def get_vxlan_dev(host):
     vxlan_port = ctl.get_alias("vxlan_port")
     vxlan_dev = "vxlan_sys_%s" % vxlan_port
-    # in asap mlnx ofed the interface name is dummy_port.
+    # In asap mlnx ofed the interface name is dummy_port.
     vxlan_dummy = "dummy_%s" % vxlan_port
-    cmd = host.run("ip l show dev %s" % vxlan_dev)
+
+    # ip link output of all interfaces of type 'vxlan'
+    cmd = host.run("ip link show type vxlan")
     out = cmd.out().strip()
-    if not out:
+    if not vxlan_dev in out:
         vxlan_dev = vxlan_dummy
```

Comment:
I don't understand why looking for a string is better than checking rc code and how does it fix an issue? This is not explained.

Reply:
Note current code not checking rc as well and probably should but checks any output exists which could cause unwanted result.
Could be you got unwanted output instead of empty output? The commit desc should explain this.

Reply:
The issue with the first suggest fix is that, when we do "ip l show dev <dev_name>" and the device is not exist, the exit code will be 1 and the test case considered failed while my goal of this check is to decide which device name exists on this machine.

Now, with this new implementation, I do listing for the vxlan device only - and it suppose to be exists, otherwise it is ok to consider the test case failed - and check if this device vxlan_sys_4789 or dummy_4789 in order to use the correct name.

Reply:
I'm not positive this is correct. You approved the first time this change worked for you and it checks first the real dev so you should have noticed it before.
Also as far as I remember return failure code from host.run() is not supposed to fail the last test.

I'll do manual testing and decide.

Reply:
According to my debug results, if specific command execution failed via host.run then this teas case will be reported as failed. 

Anyway, I am waiting your inputs.

Reply:
Any updates?